prior/__init__.py

```diff
@@ -319,10 +329,125 @@ def get_cached_sha() -> Optional[str]:
+    project_dir = f"{MODEL_DIR}/{entity}/{project}"
+    sha, token = _clone_repo(
+        base_dir=MODEL_DIR, entity=entity, project=project, revision=revision, offline=offline
+    )
+
+    git_lfs_cmd = _get_git_lfs_cmd()
+    oldpath = os.environ["PATH"]
+    try:
+        # The below PATH setting needs to happen before running any git commands as otherwise git
+        # will not see the git-lfs download which causes all sorts of weird issues.
+        if git_lfs_cmd != "git lfs":
+            # Need to set the path so that git sees git-lfs below
+            os.environ["PATH"] = f'{os.environ["PATH"]}:{os.path.dirname(git_lfs_cmd)}'
+
+        # download the dataset
+        models_path = f"{project_dir}/{sha}"
+        if not os.path.exists(models_path):
+            with LockEx(f"{project_dir}/lock"):
+                logging.debug(
+                    f"Downloading project {project} at revision {revision} to {models_path}."
+                )
+                token_prefix = f"{token}@" if token else ""
+
+                # using smudges avoid downloading all LFS files / weights
+                old_smudge_value = os.environ.get("GIT_LFS_SKIP_SMUDGE", None)
+                os.environ["GIT_LFS_SKIP_SMUDGE"] = "1"
+                subprocess.run(
+                    args=[
+                        "git",
+                        "clone",
+                        f"https://{token_prefix}github.com/{entity}/{project}.git",
+                        models_path,
+                    ],
+                    stdout=subprocess.DEVNULL,
+                    stderr=subprocess.DEVNULL,
+                )
+                if old_smudge_value is not None:
+                    os.environ["GIT_LFS_SKIP_SMUDGE"] = old_smudge_value
+
+                logging.debug(f"Downloaded model to {models_path}")
+                # change the subprocess working directory to the dataset directory
+                os.chdir(models_path)
+                subprocess.run(
+                    args=["git", "checkout", sha],
+                    stderr=subprocess.DEVNULL,
+                    stdout=subprocess.DEVNULL,
+                )
+                logging.debug(f"Checked out {sha}")
+
+                subprocess.run(
+                    args="git restore --staged .".split(),
+                    stderr=subprocess.DEVNULL,
+                    stdout=subprocess.DEVNULL,
+                )
+
+        logging.debug(f"Using project {project} at revision {revision} in {models_path}.")
+        os.chdir(models_path)
+
+        with open("models.json", "r") as f:
+            models = json.load(f)
+            if model not in models:
+                raise ValueError(f"Model ({model}) not found in {models.keys()}")
+
+        out0 = subprocess.run(
+            f"{git_lfs_cmd} install".split(),
+            stdout=subprocess.DEVNULL,
+        )
+        out1 = subprocess.run(
+            f"{git_lfs_cmd} fetch origin --include {models[model]}".split(),
```

Comment:
This is the super cool line that only downloads the specified model, and not the entire repo! 🤩 

(Setting `os.environ["GIT_LFS_SKIP_SMUDGE"] = "1"` a bit earlier makes it so that all the git lfs files are not downloaded upon cloning.)